packages/x-data-grid/src/constants/gridClasses.ts

```diff
@@ -408,6 +408,10 @@ export interface GridClasses {
    * Styles applied to the virtualization container.
    */
   virtualScroller: string;
+  /**
+   * Styles applied to the virtualization container when it is scrollable in the horizontal direction.
+   */
+  'virtualScroller--hasScrollX': string;
```

Comment:
Does this need to be public or can we `@ignore` it?

Reply:
Good shout, can keep it private https://github.com/mui/mui-x/pull/14293/commits/4303b0ebb85cb9ead86e1cab57b1dfd9d56b4182